main.go

```diff
@@ -128,8 +138,49 @@ func setupHttpClient() (*http.Client, error) {
 	return client, nil
 }
 
-func processTarGz(reader io.Reader, file string, to string) (string, error) {
+func processArchive(t ArchiveType, reader io.Reader, file string, to string) (string, error) {
+	switch t {
+	case TarGz:
+		return processTarGz(reader, file, to)
+	case Zip:
+		return processZip(reader, file, to)
+	default:
+		return "", errors.New("invalid archive type")
+	}
+}
+
+func processZip(reader io.Reader, file string, to string) (string, error) {
+	body, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return "", fmt.Errorf("failed to read: %w", err)
+	}
+
+	zipReader, err := zip.NewReader(bytes.NewReader(body), int64(len(body)))
```

Comment:
It's a bummer `zip.NewReader` needs to know the file size ahead of time. I might follow up with an enhancement that does a `HEAD` on the URL to retrieve the size and/or see if github tends to respond with `ContentLength` populated. Or you're welcome to experiment and contribute that.

Reply:
Would an extra HTTP request be better than reading everything in memory?

Reply:
I'm hoping `ContentLength` on the existing request will suffice, but worst case the `HEAD` is a cheap request. With all that said, it may not even be worth all the complexity just so the zip processing can be streamed.

Reply:
Yes, this is promising. I set a breakpoint at L94 of main and confirmed content length is populated in a github request at least:

![image](https://user-images.githubusercontent.com/988985/76686464-8ae0e000-65e9-11ea-8603-9074a0a0d8b9.png)

Reply:
nm, I see now that it wants a random access https://golang.org/pkg/io/#ReaderAt so streaming doesn't work anyway. All is good :)

Reply:
Too bad!

Reply:
I guess if people run into the need for very large zip files, then could always enhance the code to download to a temp file.